tests_integration/lib/test_case.py

```diff
@@ -0,0 +1,29 @@
+from abc import ABC
+
+from fastapi import APIRouter
+
+from tests_integration.lib.integration_test_helper import IntegrationTestHelper
+
+
+class TestCase(ABC):
+    """
+    Abstract base class for a single test case in a scenario.
```

Comment:
Can you elaborate here on this docstring what the expected lifetime of the testcase instance is? I think that we run `prepare()` then the bot does its thing, and then we run again with `validate()`, correct?

Reply:
Yes, there even is only one global instance of each test case that might run multiple times (e61bb2d641b1d8a84dfc4fc733d4c63de1a1c400).

Ideally, we would create one instance *per test case execution*, but I think this adds too much complexity for now, as we don't store any state in a test case currently.